tests: drivers: build_all: display: Add mipi-dsi devices to build test

soburi opened the pull request:
Add build tests for following devices.

- himax,hx8394
- frida,nt35510
- orisetech,otm8009a
- raydium,rm67162
- raydium,rm68200

soburi committed:
drivers: mipi_dsi: Add dummy driver for vnd,mipi-dsi

Add dummy driver for "vnd,mipi-dsi" to use in build_all tests.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

soburi committed:
tests: drivers: build_all: display: Add mipi-dsi devices to build test

Add build tests for the following devices.

- himax,hx8394
- frida,nt35510
- orisetech,otm8009a
- raydium,rm67162
- raydium,rm68200

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

soburi committed:
tests: drivers: build_all: display: Add config for `solomon,ssd1327fb`

Add configuration for "solomon,ssd1327fb" to enable build test.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>

danieldegrasse commented on tests/drivers/build_all/display/testcase.yaml:
```diff
@@ -7,4 +7,7 @@ common:
-  drivers.display.default: {}
+  drivers.display.default:
+    extra_configs:
+      - CONFIG_DISPLAY_NT35510_INIT_PRIORITY=87
```
Why do we need to set these explicitly?

soburi replied:
A display connected by DSI is needed to initialize later than the DSI driver (MIPI_DSI_INIT_PRIORITY=86).

danieldegrasse replied:
Ok, makes sense- is there a reason these displays use an init priority lower than the DSI driver currently though? It seems like they shouldn't have a default lower than the DSI driver default

danieldegrasse replied:
Looking at this more, the default DSI init priority is 86, and default display init priority is 85. @gmarull do you recall the reason that DSI controllers were setup to initialize after displays by default?

gmarull replied:
I don’t know, but I guess it can be adjusted, now priorities are checked with @fabiobaltieri tool

fabiobaltieri replied:
How about changing one of the two to match the other? That way the sub-priority kicks in and takes care of all this mess automatically.

danieldegrasse replied:
Sure, seems reasonable- I didn't know we were using ordinals now for sub-priorities, that is great to see! @soburi, would you like to change the DSI init priority to 85 as part of this PR, or should we split that fix into a separate PR?

fabiobaltieri replied:
Yeah the change went a bit under the radar but it's massive, basically you could squash all devicetree based drivers (i.e. all of them) to the same number and it would initialize in the right sequence, I was even tempted to propose removing the priority entirely from the DT_DEVICE macros, problem is if you want to sneak in some non-dt driver in a specific spot in the sequence.

soburi replied:
> Sure, seems reasonable- I didn't know we were using ordinals now for sub-priorities, that is great to see! @soburi, would you like to change the DSI init priority to 85 as part of this PR, or should we split that fix into a separate PR?

I applied it. Please confirm the changes.

soburi committed:
drivers: mipi_dsi: Change MIPI_DSI priority to the same as DISPLAY

These priorities can be resolved automatically with the sub-priority
mechanism, so set them to the same value.

Signed-off-by: TOKITA Hiroshi <tokita.hiroshi@gmail.com>
